dm/dm/worker/subtask.go

```diff
@@ -588,7 +588,7 @@ func (st *SubTask) Resume(relay relay.Process) error {
 		return nil
 	}
 
-	if !st.stageCAS(pb.Stage_Paused, pb.Stage_Resuming) {
+	if !st.stageCAS(pb.Stage_Paused, pb.Stage_Resuming) && !st.stageCAS(pb.Stage_Stopped, pb.Stage_Resuming) {
```

Comment:
this will fail if between two "CAS" `Stage_Stopped` becomes `Stage_Paused`. The check should atomic.

Reply:
use atomic check in  https://github.com/pingcap/tiflow/pull/4510/commits/8955c534dc224cdf6457ea84b824d94c1f5df6b3